### PR TITLE
Apply RemoveTestPrefix

### PR DESCRIPTION
**Pull request by koppor**
Follow-up to https://github.com/JabRef/jabref/pull/10797 and https://github.com/JabRef/jabref/pull/10799.

This applies [Remove test prefix from JUnit 5 tests](https://docs.openrewrite.org/recipes/java/testing/cleanup/removetestprefix) and fixes the issues caused by https://github.com/openrewrite/rewrite-testing-frameworks/issues/462 manually.

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by koppor** [comment]
Only FetcherTests fail, because of CiteSeer and Grobid (and others)

File: src/test/java/org/jabref/logic/layout/format/RTFCharsTest.java [hunk 1]
```diff
@@ -180,13 +180,13 @@ void testSpecialCharacters() {
...
     })
-    public void testMoreSpecialCharacters(String specialChar, String expectedResult) {
+    public void moreSpecialCharacters(String specialChar, String expectedResult) {
         String formattedStr = formatter.format(specialChar);
         assertEquals(expectedResult, formattedStr);
     }
 
     @Test
-    void testRTFCharacters() {
+    void rTFCharacters() {
```
Comment (calixtus) [inline src/test/java/org/jabref/logic/layout/format/RTFCharsTest.java:189]: ,,,?

**Review by calixtus — CHANGES_REQUESTED** [review]
one misspelled method name

**Comment by Siedlerchr** [comment]
I don't like this  refactoring, Even though it's not necessary, I find it easier to have the methods with test prefixed, helps it to find them

**Comment by calixtus** [comment]
I have no strong opinion here, but it should be consistent everywhere. For now we have methods with and some without the prefix.

About the argument: I think the ide should help here. In intellij test classes have a green bg. But I don't know if this is what you mean?

What's more of interest to me in this matter is: what recommendation does junit make and how do the error messages look if junit produces some.

File: src/test/java/org/jabref/logic/layout/format/RTFCharsTest.java (hunk 1 above)
Comment (koppor) [reply src/test/java/org/jabref/logic/layout/format/RTFCharsTest.java:189]: Submitted https://github.com/openrewrite/rewrite-testing-frameworks/issues/473

**Comment by koppor** [comment]
+1 for consistency. Therefore, I applied it.

It is also based on the experience by our friends at Moderne. Thus, I trust them --> https://docs.openrewrite.org/recipes/java/testing/cleanup/removetestprefix

**Review by Siedlerchr — DISMISSED** [review]

**Comment by Siedlerchr** [comment]
There is something wrong with the lobid fetcher:

Failed to map supported failure 'org.opentest4j.AssertionFailedError: expected: <Optional[Paper Title]> but was: <Optional.empty>' with mapper 'org.gradle.api.internal.tasks.testing.failure.mappers.OpenTestAssertionFailedMapper@151d4a62': Cannot invoke "Object.getClass()" because "expectedValue" is null

**Comment by koppor** [comment]
This PR did not touch the LOBID fetcher:

image

**Review by calixtus — APPROVED** [review]

**Pull request merged**